### ISSUE TITLE
ILSVRC12 Caffe hosted mean, different than calculated mean

### ISSUE THREAD

**Issue by deanpospisil** [BVLC/caffe#4482]
I've been having trouble replicating the training of @jeffdonahue [reference performance](https://www.google.com/url?q=https%3A%2F%2Fgist.github.com%2Fjeffdonahue%2Fb14f9611442ce699eb5c.&sa=D&sntz=1&usg=AFQjCNH_0U1Z0AlVrih5kOFgszZ1_hDGvQ)
following: [Caffe instructions for CaffeNet](http://caffe.berkeleyvision.org/gathered/examples/imagenet.html)
One thing I checked was whether the mean that I calculated on ILSCVR12 (using make_imagenet_mean) was the same as that hosted by caffe: [aux data download](http://dl.caffe.berkeleyvision.org/caffe_ilsvrc12.tar.gz). It was not (mean difference between my images and theirs=0.069, mean pixel value across images: 114), so small but significant.

Using the exact prototxts from [caffenet reference on github](https://github.com/BVLC/caffe/tree/master/models/bvlc_reference_caffenet) I trained with my mean and then the hosted mean:
![my_mean_vs_hosted_mean](https://cloud.githubusercontent.com/assets/15167171/16929207/865ff810-4ceb-11e6-81e6-b2c9d60d526e.jpg)
In blue, is the correct mean where it initializes and runs quite similarly to the reference training by Donahue, whereas in green using the hosted mean, it takes a little while for it to get going.
This might be the source of the problem referenced in:
https://github.com/BVLC/caffe/issues/430, where the workaround was to decrease the biases.
and 
https://github.com/BVLC/caffe/issues/3243 where the workaround was to try a different initialization.

I'm curious how this mean might have ended up being different, if there is some difference in how it is calculated or if ILSCVR12 images changed, or if there is some pre-processing step that was done by Donahue, that might explain this difference. 
I know GitHub isn't the place to discuss training issues, but I thought it would be helpful to make sure the correct mean is being hosted. I also have posted on the Caffe forum and haven't made any progress there https://groups.google.com/forum/#!topic/caffe-users/s81jvLLjvuk.
Perhaps the difference in means might point me to a difference in the training sets that is preventing me from getting the performance of Donahue, as you see my training diverges, from Donahues, and I have seen this for many different initializations, even when waiting many more iterations. Below are some examples where I varied batch size:
![training_problems](https://cloud.githubusercontent.com/assets/15167171/16929621/6fffbcac-4ced-11e6-9360-649d5497c538.jpeg)
Happy to provide more detail

**Comment by seanbell** [comment]
Did you fix the random seed?  That will have a way bigger effect than a small difference in the mean.

**Comment by deanpospisil** [comment]
Woops! I did not fix the seed. 
I ran a quick test with the seed fixed, and it would seem your suspicions were correct:
![same_seed_different_means](https://cloud.githubusercontent.com/assets/15167171/16970010/eec88c0e-4dcd-11e6-8a7f-6e0ccc772d1a.png)
Not sure why but with the calculated mean the training breaks out of that initial lull faster. It'd be interesting to see if that happens on average, and what about the initialization gets it through faster (weights more negative or positive)...

Any suggestions as to what might be preventing me from replicating the training of Donahue? Is training sensitive enough to initialization that I might need to try many more times before I get a good looking curve like his?

**Comment by fancyerii** [comment]
I confront exactly the same problem. have you solved this problem now?

**Comment by fancyerii** [comment]
If I use hosted mean file, it get accuracy 58%. if use my own calculated mean file, it can not be well trained. only get random 0.001 accuracy and loss don't decrease
